source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -384,27 +383,42 @@ void CacheFilter::processSuccessfulValidation(Http::ResponseHeaderMap& response_
     return Http::HeaderMap::Iterate::Continue;
   });
 
-  if (should_update_cached_entry) {
-    // TODO(yosrym93): else the cached entry should be deleted.
-    // Update metadata associated with the cached response. Right now this is only response_time;
-    const ResponseMetadata metadata = {time_source_.systemTime()};
-    cache_.updateHeaders(*lookup_, response_headers, metadata);
-  }
+  // Update metadata associated with the cached response. Right now this is only response_time;
+  const ResponseMetadata metadata = {time_source_.systemTime()};
+  cache_.updateHeaders(*lookup_, response_headers, metadata);
 
   // A cache entry was successfully validated -> encode cached body and trailers.
   encodeCachedResponse();
 }
 
 // TODO(yosrym93): Write a test that exercises this when SimpleHttpCache implements updateHeaders
-bool CacheFilter::shouldUpdateCachedEntry(const Http::ResponseHeaderMap& response_headers) const {
-  ASSERT(isResponseNotModified(response_headers),
-         "shouldUpdateCachedEntry must only be called with 304 responses");
-  ASSERT(lookup_result_, "shouldUpdateCachedEntry precondition unsatisfied: lookup_result_ "
+bool CacheFilter::isSuccessfulValidation(const Http::ResponseHeaderMap& response_headers) const {
+  ASSERT(lookup_result_, "isSuccessfulValidation precondition unsatisfied: lookup_result_ "
                          "does not point to a cache lookup result");
   ASSERT(filter_state_ == FilterState::ValidatingCachedResponse,
-         "shouldUpdateCachedEntry precondition unsatisfied: the "
+         "isSuccessfulValidation precondition unsatisfied: the "
          "CacheFilter is not validating a cache lookup result");
+  // Should only update for 304 responses
+  if (!isResponseNotModified(response_headers)) {
+    return false;
+  }
 
+  // Delete the entry instead of updating if
```

Comment:
I'm surprised by this reference to "delete" in a const method which returns bool. I guess the entry is deleted if this method returns false? Perhaps this comment could say:

"This entry is not valid if: "

Reply:
That makes sense. Thanks for pointing it out :D We're not deleting it, we are just treating it as a failed validation. A later insert should overwrite it.